src/python/grpcio_testing/grpc_testing/_channel/_rpc_state.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+import grpc
+from grpc_testing import _common
+
+
+class State(_common.ChannelRpcHandler):
+
+    def __init__(self, invocation_metadata, requests, requests_closed):
+        self._condition = threading.Condition()
+        self._invocation_metadata = invocation_metadata
```

Comment:
It could, but there's no need for it to be. Each of the three [metadata objects](https://grpc.io/grpc/python/grpc.html#glossary) in an RPC is treated as a (probably immutable) [sequence](https://docs.python.org/2/library/stdtypes.html#sequence-types-str-unicode-list-tuple-bytearray-buffer-xrange). Metadata is accepted whole from the application on one side, delivered whole to the application on the other side, and mostly not molested by the gRPC code in between.

In the particular circumstance of this instance-scope field: non-`None` indicating metadata accepted from the system under test and not yet delivered to the testing system and `None` indicating metadata having already been delivered to the testing system are exactly the semantics we want. We want semantics simpler than one usually gets with a mutable list-valued field.